bats/tests/k8s/up-downgrade-k8s.bats

```diff
@@ -3,13 +3,32 @@
 load '../helpers/load'
 ARCH_FOR_KUBERLR=amd64
 
+local_setup_file() {
+    if semver_eq "$RD_KUBERNETES_VERSION" "$RD_KUBERNETES_ALT_VERSION"; then
+        printf "Cannot upgrade from %s to %s\n" \
+            "$RD_KUBERNETES_VERSION" "$RD_KUBERNETES_ALT_VERSION" |
+            fail
+    fi
+    # It is undefined whether RD_KUBERNETES_VERSION is greater or less than
+    # RD_KUBERNETES_ALT_VERSION (and it's expected to flip in CI); actually
+    # compare them so we can expect to wipe data on the downgrade.
+    if semver_gt "$RD_KUBERNETES_VERSION" "$RD_KUBERNETES_ALT_VERSION"; then
+        export RD_KUBERNETES_VERSION_LOW=$RD_KUBERNETES_ALT_VERSION
+        export RD_KUBERNETES_VERSION_HIGH=$RD_KUBERNETES_VERSION
+    else
+        export RD_KUBERNETES_VERSION_LOW=$RD_KUBERNETES_VERSION
+        export RD_KUBERNETES_VERSION_HIGH=$RD_KUBERNETES_ALT_VERSION
+    fi
+}
+
 @test 'factory reset' {
     factory_reset
 }
 
 @test 'start rancher desktop' {
-    start_kubernetes
-    wait_for_kubelet
+    # Force use the pre-upgrade version
+    RD_KUBERNETES_VERSION=$RD_KUBERNETES_VERSION_LOW start_kubernetes
+    wait_for_kubelet "$RD_KUBERNETES_VERSION_LOW"
```

Comment:
I find this easier to read:

```suggestion
    # Force use the pre-upgrade version
    RD_KUBERNETES_VERSION=$RD_KUBERNETES_VERSION_LOW
    start_kubernetes
    wait_for_kubelet
```

Reply:
I am limiting it to the single lines to ensure that nothing else will pick up the version override (for when we add extra lines later).

Reply:
> I am limiting it to the single lines to ensure that nothing else will pick up the version override

Why would you not want that? I think it would be good to be able to assume that `RD_KUBERNETES_VERSION` is the currently running Kubernetes version. See `foreach_k3s_version`, that sets `RD_KUBERNETES_VERSION` to each version in turn (it couldn't even work otherwise). So any helper function that needs to know has to assume that `RD_KUBERNETES_VERSION` is the current version.

But anyways, it doesn't matter unless/until you add more lines to the test.